wd_helper.au3

```diff
@@ -636,6 +636,199 @@ Func _WD_FrameLeave($sSession)
+					$iExt = @extended
+					$iErr = $_WD_ERROR_InvalidExpression
+					ExitLoop
+				Else ; StringRegExp = 0 no MATCH
+					_ArrayDelete($a_Result, $i)
+				EndIf
+			Next
+			If UBound($a_Result) = 0 Then
+				$iErr = $_WD_ERROR_NotFound
+			EndIf
+		EndIf
+
+		; select desired DataType for the $vResult
+		If $bReturnAsArray Or $bReturnHTML Then ; do not return huge amount of HTML data to string - only as array as they can contain restricted data
+			$vResult = $a_Result
+		Else ; string is option for testing and asking support thus should not return huge amount of HTML data to string as they can contain restricted data
+			$vResult = _ArrayToString($a_Result)
+		EndIf
+	EndIf
+
+	; Back to "calling frame"
+	If $sStartLocation Then
+		_WD_FrameEnter($sSession, $sStartLocation)
+		$iErr = @error
+	EndIf
+
+	If $iErr And $iErr <> $_WD_ERROR_InvalidExpression Then
+		$sMessage = 'Was not able to check / back to "calling frame"'
+	EndIf
+
+	#EndRegion - _WD_FrameList - post processing
+
+	$sMessage = ($sMessage And $_WD_DEBUG = $_WD_DEBUG_Full) ? (' Information: ' & $sMessage) : ("")
+	Return SetError(__WD_Error($sFuncName, $iErr, $sParameters & $sMessage), $iExt, $vResult)
+EndFunc   ;==>_WD_FrameList
+
+; #INTERNAL_USE_ONLY# ===========================================================================================================
+; Name ..........: __WD_FrameList_Internal
+; Description ...: function that is used internally in _WD_FrameList, even recursively when nested frames are available
+; Syntax ........: __WD_FrameList_Internal($sSession, $sLevel, $sFrameAttributes, $bReturnHTML)
+; Parameters ....: $sSession            - Session ID from _WD_CreateSession
+;                  $sLevel              - frame location level ... path
+;                  $sFrameAttributes    - all <iframe ....> atributes
+;                  $bReturnHTML         - On true function will return frame document HTML source
+; Return values .: Success - array or string
+;                  Failure - "" (empty string) and sets @error to one of the following values:
+;                  - $_WD_ERROR_Exception
+;                  - $_WD_ERROR_InvalidExpression
+; Author ........: mLipok
+; Modified ......:
+; Remarks .......:
+; Related .......: _WD_FrameList, _WD_FrameEnter, _WD_GetFrameCount
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func __WD_FrameList_Internal($sSession, $sLevel, $sFrameAttributes, $bReturnHTML)
+	Local Const $sFuncName = "__WD_FrameList_Internal"
+	Local Const $sParameters = 'Parameters:    Level=' & $sLevel & '   ReturnHTML=' & $bReturnHTML ; intentionally $sFrameAttributes is not listed here to not put too many data into the log
+	Local $iErr = $_WD_ERROR_Success
+	Local $vResult = '', $s_URL = '', $s_HTML = '', $sMessage = ''
+
+	Local $a_Level = StringSplit($sLevel, '/')
```

Comment:
as one of further step this part can be replaced to `_WD_FrameEnter($sSession, $sLevel)`

but after rethinking at first place this 
https://github.com/Danp2/au3WebDriver/pull/363
and 
https://github.com/Danp2/au3WebDriver/pull/364

(in the given order)

PR should be reviewed and merged

Reply:
suplemented in:
[further changes](https://github.com/Danp2/au3WebDriver/pull/362/commits/3bdf5437e1f1792e68f9d69804f593ce4cd6ac52)